src/utils/inputValidation.ts

```diff
@@ -0,0 +1,21 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { valueOnTimeout } from "./timeout";
+
+const inputValidationTimeoutMs = 2000;
+
+/**
+ * Intended to be used for VS Code validateInput to protect against long-running validations. If a time-out occurs or the action throws,
+ * returns undefined (indicating a valid input). Use for optional validations.
+ */
+export function validOnTimeoutOrException(inputValidation: () => Promise<string | undefined>, timeoutMs?: number): Promise<string | undefined> {
+    try {
+        timeoutMs = timeoutMs || inputValidationTimeoutMs;
```

Comment:
Is there any chance someone wants timeoutMs to be 0? If so, this code would ignore that and set the timeout to 2000.

Reply:
Doesn't seem like a reasonable scenario to me.

Reply:
Okey dokey artichokey